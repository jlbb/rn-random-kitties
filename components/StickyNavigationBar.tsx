import { StyleSheet, View } from "react-native";
import React from "react";

import { colors } from "../config/contants";

interface StickyNavigationBarProps {
  children: any;
}

const StickyNavigationBar = ({
  children
}: StickyNavigationBarProps): JSX.Element => {
  return <View style={styles.container}>{children}</View>;
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.WHITE_DEFAULT,
    flexDirection: "row",
    height: 70,
    position: "absolute",
    top: 0,
    width: "100%",
    zIndex: 2
  }
});

export default StickyNavigationBar;
