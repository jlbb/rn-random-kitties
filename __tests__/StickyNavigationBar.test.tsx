import StickyNavigationBar from "../components/StickyNavigationBar";
import { shallow } from "enzyme";

describe("StickyNavigationBar", () => {
  it("renders correctly", () => {
    expect(
      shallow(<StickyNavigationBar>{"Button here"}</StickyNavigationBar>)
    ).toMatchSnapshot();
  });
});
