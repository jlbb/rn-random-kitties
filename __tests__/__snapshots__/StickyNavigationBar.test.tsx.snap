// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickyNavigationBar renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flexDirection": "row",
      "height": 70,
      "position": "absolute",
      "top": 0,
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  Button here
</View>
`;
